pkg/network/usm/tests/tracer_usm_kafka_classification_linux_test.go

```diff
@@ -270,13 +284,15 @@ func testKafkaProtocolClassification(t *testing.T, tr *tracer.Tracer, clientHost
 		},
 	}
 
-	for produceVersion := kafka.MinSupportedProduceRequestApiVersion; produceVersion <= kafka.MaxSupportedProduceRequestApiVersion; produceVersion++ {
+	// Generate tests for all support Produce versions + 1 unsupported version
+	for produceVersion := kafka.ClassificationMinSupportedProduceRequestApiVersion; produceVersion <= kafka.ClassificationMaxSupportedProduceRequestApiVersion+1; produceVersion++ {
```

Comment:
why the loop iterates until `+1` from the max supportedProduceRequest?
With your change, `ClassificationMaxSupportedProduceRequestApiVersion` reaches to the last api version (12) of produce, so there is a spare, and irrelevant test here (that should fail in the CI and local runs)

Reply:
if you try to force the kafka client library to use `produce v13` (which does not exist), it will use `v12` (you can see that using wireshark)
so for produce it just means we run the test twice for v12

same will happen once we finish upgrading our classification to support all known fetch versions 

instead, you can use `version.LookupMaxKeyVersion(produceAPIKey)` to get the maximum supported version of produce (and fetch) by the client library

then iterate until `max(version.LookupMaxKeyVersion(produceAPIKey), kafka.ClassificationMaxSupportedProduceRequestApiVersion+1)`